distributed/security.py

```diff
@@ -1,18 +1,107 @@
+from __future__ import annotations
+
 import datetime
 import os
+import sys
 import tempfile
 
 try:
     import ssl
 except ImportError:
     ssl = None  # type: ignore
 
+try:
+    from ssl import OPENSSL_VERSION_INFO as _OPENSSL_VERSION_INFO
+except ImportError:
+    _OPENSSL_VERSION_INFO = (0, 0, 0, 0, 0)
+
+from typing import TYPE_CHECKING
+
 import dask
 from dask.widgets import get_template
 
+if TYPE_CHECKING:
+    from _typeshed import StrOrBytesPath
+
 __all__ = ("Security",)
 
 
+if sys.version_info >= (3, 10) or _OPENSSL_VERSION_INFO >= (1, 1, 0, 7):
```

Comment:
Note if this is using newer APIs from OpenSSL, Python 3.10 is the only version I'm aware of doing that (from a different context). So checking the OpenSSL version may not be sufficient. In fact it may lead to issues as packagers may build with a newer OpenSSL even with old Python versions (as is the case in conda-forge)

Reply:
This should work with both newer openssl versions or newer python versions